File: modules/app/src/main/java/org/artembogomolova/demo/webapp/filter/AnonymousAuthenticationFilterImpl.java
package org.artembogomolova.demo.webapp.filter;

import javax.servlet.http.HttpServletRequest;
import org.artembogomolova.demo.webapp.dao.service.UserRepoService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.Authentication;
import org.springframework.security.web.authentication.AnonymousAuthenticationFilter;
import org.springframework.stereotype.Component;

@Component
public class AnonymousAuthenticationFilterImpl extends AnonymousAuthenticationFilter {

  private transient final UserRepoService userRepoService;

  @Autowired
  public AnonymousAuthenticationFilterImpl(UserRepoService userRepoService) {
    super("anonymous");
    this.userRepoService = userRepoService;
  }

  @Override
  protected Authentication createAuthentication(HttpServletRequest request) {
    return userRepoService.getGuestUserToken();
  }
}

File: app/src/main/kotlin/org/artembogomolova/demo/webapp/main/filter/AnonymousAuthenticationFilterImpl.kt
package org.artembogomolova.demo.webapp.main.filter

import javax.servlet.http.HttpServletRequest
import org.artembogomolova.demo.webapp.main.dao.service.UserRepoService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.core.Authentication
import org.springframework.security.web.authentication.AnonymousAuthenticationFilter
import org.springframework.stereotype.Component

@Component
class AnonymousAuthenticationFilterImpl @Autowired constructor(
    private val userRepoService: UserRepoService
) : AnonymousAuthenticationFilter("anonymous") {

    override fun createAuthentication(request: HttpServletRequest): Authentication {
        return userRepoService.guestUserToken
    }
}